date_range/models/date_range_type.py

```diff
@@ -83,7 +83,8 @@ def _check_company_id(self):
                     continue
                 if bool(
                     rec.date_range_ids.filtered(
-                        lambda r: r.company_id and r.company_id != rec.company_id
+                        lambda r, drt=rec: r.company_id
```

Comment:
This violates B023, `any([range_date.company_id != rec.company_id for range_date in rec.date_range_ids])` can make the ruff lint happy

Reply:
hmm, somehow the pre-commit could pass :thinking: